infinitic-storage-mysql/src/main/kotlin/io/infinitic/storage/mysql/MySQL.kt

```diff
@@ -0,0 +1,36 @@
+/**
+ * "Commons Clause" License Condition v1.0
+ *
+ * The Software is provided to you by the Licensor under the License, as defined
+ * below, subject to the following condition.
+ *
+ * Without limiting other conditions in the License, the grant of rights under the
+ * License will not include, and the License does not grant to you, the right to
+ * Sell the Software.
+ *
+ * For purposes of the foregoing, “Sell” means practicing any or all of the rights
+ * granted to you under the License to provide to third parties, for a fee or
+ * other consideration (including without limitation fees for hosting or
+ * consulting/ support services related to the Software), a product or service
+ * whose value derives, entirely or substantially, from the functionality of the
+ * Software. Any license notice or attribution required by the License must also
+ * include this Commons Clause License Condition notice.
+ *
+ * Software: Infinitic
+ *
+ * License: MIT License (https://opensource.org/licenses/MIT)
+ *
+ * Licensor: infinitic.io
+ */
+
+package io.infinitic.storage.mysql
+
+import com.sksamuel.hoplite.Secret
+
+data class MySQL(
+    val host: String = "127.0.0.1",
+    var port: Int = 3306,
+    var user: String? = null,
+    var password: Secret? = null,
+    var database: String? = null
+)
```

Comment:
If you want Hoplite library provide a direct config object for HikariDatasource not sure it's necessary.
But it may allow user to use config to set specific settings if necessary. 
And it may allow to use all JDBC database supported by Hikari directly (if all the SQL statement are compatible)
Here is link to Hoplite doc:
https://github.com/sksamuel/hoplite#:~:text=hoplite%2Dhikaricp%20module-,HikariDataSource,-Converts%20nested%20config
In yaml config file then you can have :
```yaml
hikariDataSource:
  jdbcUrl: ${MYSQL_JDBC_URL}
  username: ${MYSQL_USER}
  password: ${MYSQL_PASSWORD}
  maxLifetime: ${MYSQL_MAX_LIFETIME}
  maximumPoolSize: ${MYSQL_MAX_POOL_SIZE}
```

Reply:
I think allowing specific & thourough configuration is a great idea as people can do whatever fits their needs best ! 

Although it can be confusing because even if you can configure you're HikariDataSource, under the hood you can't actually use any driver supported because we use very specific statements (like upserts INSERT INTO ... ON DUPLICATE KEY UPDATE ...) which are not RDBMS agnostic (it won't work in PostgreSQL for example) to be able to use any JDBC drivers we would need to have only sets of simple statement (like SELECT then UPDATE or INSERT). 
This means more code & less performance but maybe it's worth it ?